modules/lfstransfer/main.go

```diff
@@ -0,0 +1,73 @@
+// Copyright 2024 The Gitea Authors. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+package lfstransfer
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"strings"
+
+	db_model "code.gitea.io/gitea/models/db"
+	repo_model "code.gitea.io/gitea/models/repo"
+	"code.gitea.io/gitea/modules/lfs"
+	"code.gitea.io/gitea/modules/lfstransfer/backend"
+	"code.gitea.io/gitea/modules/lfstransfer/transfer"
+	"code.gitea.io/gitea/modules/log"
+	"code.gitea.io/gitea/modules/setting"
+	"code.gitea.io/gitea/modules/storage"
+)
+
+func initServices(ctx context.Context) error {
+	setting.MustInstalled()
+	setting.LoadDBSetting()
```

Comment:
I don't think it is good to connect to DB and LFS store directly. In git operations we are using internal rest API to access resources like database etc. Otherwise we will have problem with sqlite database locking etc

Reply:
@lafriks I saw functions like `initDB()` that multiple commands seem to call:
https://github.com/go-gitea/gitea/blob/08579d6cbb65399dec408f3b90bc9a9e285e6206/cmd/cmd.go#L60
so I thought this was fine, and the underlying implementation was race-free or something.
But now that you mention it, looking at the call graph... most of them do seem to be maintainance commands etc. that don't run concurrently with the main process :P

So the preferred way of doing this would be a REST API? Well I suppose ait's a good thing that one already exists.

Reply:
yes there is already one in place that uses secret in app.ini for authorization. Other cli tools that connect directly to database are just for administration (so used only rarely) but if used form ssh process this can cause race/locking problems

Reply:
@lafriks I rewrote the backends to just proxy calls to the (public) LFS API.

Pushes and clones work... but there's sometimes (by which I mean decently reproducible) bug where _after_ transferring and verification and everything... the connection just hangs for almost exactly 120 seconds. After which it just ends gracefully and git proceeds with its push.

It's particularly nasty, I have no leads on it yet. The best I can figure out is that it looks like the `git-lfs` client seems to not send the quit command that it should at the end of the protocol, or that it isn't coming through.